test/CTestList.cmake

```diff
@@ -37,19 +37,21 @@ function(add_test_r TEST_NAME)
     if(AMR_WIND_ENABLE_CUDA)
       set(FCOMPARE_TOLERANCE "-r 1e-10 --abs_tol 1.0e-12")
     endif()
-    # Use fcompare to test diffs in plots against gold files
-    if(AMR_WIND_TEST_WITH_FCOMPARE)
-      set(FCOMPARE_COMMAND "&& ${FCOMPARE_EXE} ${FCOMPARE_TOLERANCE} ${PLOT_GOLD} ${PLOT_TEST}")
-    endif()
     if(AMR_WIND_ENABLE_MPI)
       set(NP 4)
       set(MPI_COMMANDS "${MPIEXEC_EXECUTABLE} ${MPIEXEC_NUMPROC_FLAG} ${NP} ${MPIEXEC_PREFLAGS}")
+      set(FCOMPARE_COMMAND "${MPIEXEC_EXECUTABLE} ${MPIEXEC_NUMPROC_FLAG} 1 ${MPIEXEC_PREFLAGS} ${FCOMPARE_EXE} ${FCOMPARE_TOLERANCE} ${PLOT_GOLD} ${PLOT_TEST}")
     else()
       set(NP 1)
       unset(MPI_COMMANDS)
+      set(FCOMPARE_COMMAND "${FCOMPARE_EXE} ${FCOMPARE_TOLERANCE} ${PLOT_GOLD} ${PLOT_TEST}")
+    endif()
+    if (AMR_WIND_TEST_WITH_FCOMPARE)
+      # Add test and actual test commands to CTest database
+      add_test(${TEST_NAME} sh -c "${MPI_COMMANDS} ${CMAKE_BINARY_DIR}/${amr_wind_exe_name} ${MPIEXEC_POSTFLAGS} ${CURRENT_TEST_BINARY_DIR}/${TEST_NAME}.i ${RUNTIME_OPTIONS} > ${TEST_NAME}.log && ${FCOMPARE_COMMAND}")
```

Comment:
So I gather `fcompare` needs an `mpirun`, but I don't think I like the duplication of this line. Seems like it should go in the lines above. Is it done to not obscure the `&&`?

Reply:
Yeah, I don't like the duplication. This function is already kind of a mess. I didn't want to do nested if conditions, so this was the best way to introduce fewest lines of change. If you see a better way, go for it.